optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -831,7 +587,175 @@ def forward(
         return BaseModelOutput(last_hidden_state=last_hidden_state)
 
 
-class ORTModelForConditionalGeneration(ORTParentMixin, ORTModel):
+class ORTDecoderForSeq2Seq(ORTSessionMixin):
+    """Decoder of an encoder-decoder model for ONNX Runtime inference."""
+
+    main_input_name = "input_ids"
+
+    def __init__(
+        self,
+        config: PretrainedConfig,
+        session: InferenceSession,
+        use_io_binding: bool | None = None,
+    ):
+        self.config = config
+        self.initialize_ort_attributes(session, use_io_binding)
+
+        self.key_value_input_names = [key for key in self.input_names if (".key" in key) or (".value" in key)]
+        self.key_value_output_names = [key for key in self.output_names if (".key" in key) or (".value" in key)]
+        self.self_attention_output_names = [key for key in self.key_value_output_names if "encoder" not in key]
+        self.cross_attention_output_names = [key for key in self.key_value_output_names if "encoder" in key]
+        self.can_use_cache = len(self.key_value_input_names) > 0 and len(self.key_value_output_names) > 0
+        self.is_merged = "use_cache_branch" in self.input_names
+
+        if self.config.model_type == "pix2struct":
+            self.vocab_size = getattr(self.config, "text_config", self.config).vocab_size
+            self.num_attention_heads = getattr(self.config, "text_config", self.config).num_heads
+            self.embed_size_per_head = (
+                getattr(self.config, "text_config", self.config).hidden_size // self.num_attention_heads
+            )
+        else:
+            self.vocab_size = getattr(self.config, "decoder", self.config).vocab_size
+            self.num_attention_heads = getattr(self.config, "decoder", self.config).num_attention_heads
```

Comment:
does this mean that we don't need a normalized_config anymore ? (uniformization for  model's config parameters happened before the min supported transformers version)

Reply:
yeah no need to use normalized configs, especially in modeling code where only very few key parameters are used as it creates a dependency on optimum's normalized configs manager in modeling. Many contributors get overwhelmed by the fact they have to open two PRs in two different packages to add a model.